src/Appwrite/Platform/Modules/Proxy/Http/Rules/Create.php

```diff
@@ -0,0 +1,198 @@
+    }
+
+    public function __construct()
+    {
+        $this
+            ->setHttpMethod(Action::HTTP_REQUEST_METHOD_POST)
+            ->setHttpPath('/v1/proxy/rules')
+            ->groups(['api', 'proxy'])
+            ->desc('Create rule')
+            ->label('scope', 'rules.write')
+            ->label('event', 'rules.[ruleId].create')
+            ->label('audits.event', 'rule.create')
+            ->label('audits.resource', 'rule/{response.$id}')
+            ->label('sdk', new Method(
+                namespace: 'proxy',
+                name: 'createRule',
+                description: <<<EOT
+                Create a new proxy rule.
+                EOT,
+                auth: [AuthType::ADMIN],
+                responses: [
+                    new SDKResponse(
+                        code: Response::STATUS_CODE_CREATED,
+                        model: Response::MODEL_PROXY_RULE,
+                    )
+                ]
+            ))
+            ->label('abuse-limit', 10)
+            ->label('abuse-key', 'userId:{userId}, url:{url}')
+            ->label('abuse-time', 60)
+            ->param('domain', null, new ValidatorDomain(), 'Domain name.')
+            ->param('resourceType', null, new WhiteList(['api', 'function', 'site']), 'Action definition for the rule. Possible values are "api", "function" and "site"')
+            ->param('resourceId', '', new UID(), 'ID of resource for the action type. If resourceType is "api", leave empty. If resourceType is "function", provide ID of the function.', true)
+            ->inject('response')
+            ->inject('project')
+            ->inject('queueForCertificates')
+            ->inject('queueForEvents')
+            ->inject('dbForPlatform')
+            ->inject('dbForProject')
+            ->callback([$this, 'action']);
+    }
+
+    public function action(string $domain, string $resourceType, string $resourceId, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform, Database $dbForProject)
+    {
+        $mainDomain = System::getEnv('_APP_DOMAIN', '');
+        if ($domain === $mainDomain) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your main domain to specific resource. Please use subdomain or a different domain.');
+        }
+
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+
+        if ($domain === $functionsDomain || $domain === $sitesDomain) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your functions or sites domain to a specific resource. Please use a different domain.');
+        }
+
+        if ($domain === 'localhost' || $domain === APP_HOSTNAME_INTERNAL) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'This domain name is not allowed. Please pick another one.');
+        }
```

Comment:
Let's combine all of this. pseudo code:

```
$denylist = [mainDomain, sitesDomain, functionsDomain, localost, APP_HOSTNAME_INTERNAL]

if denyList includes $domain
	throw exception
```

For all, we can give generic `This domain name is not allowed. Please pick another one.`